specificatie/genereervariant/openapi.yaml

```diff
@@ -3036,13 +3622,109 @@ components:
+      description: De Link Object zoals gespecificeerd in https://tools.ietf.org/html/draft-kelly-json-hal-08#section-5;
+        Deze link kan als templated link worden aangeboden. [URI-templating is hier
+        beschreven](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.2.0/features/uri-templating.feature).
     Href:
       type: string
       example: https://datapunt.voorbeeldgemeente.nl/api/v{major-versie}/resourcename/{resource-identificatie}
     HalPaginationLinksMetLast:
+      description: |
+        HalPaginationLinksMetLast bevat de link properties die nodig zijn om te kunnen navigeren in een eindige collectie:
+
+        - first: uri voor het opvragen van de eerste pagina van deze collectie
+        - previous: uri voor het opvragen van de vorige pagina van deze collectie
+        - next: uri voor het opvragen van de volgende pagina van deze collectie
+        - last: uri voor het opvragen van de laatste pagina van deze collectie
+      example:
+        self:
+          href: /resourcenaam?page=4
+        first:
+          href: /resourcenaam?page=1
+          title: Eerste pagina
+        previous:
+          href: /resourcenaam?page=3
+          title: Vorige pagina
+        next:
+          href: /resourcenaam?page=5
+          title: Volgende pagina
+        last:
+          href: /resourcenaam?page=8
+          title: Laatste pagina
       allOf:
       - $ref: '#/components/schemas/HalPaginationLinks'
+      - type: object
+        properties:
+          last:
+            $ref: '#/components/schemas/HalLink'
+    HalPaginationLinks:
+      description: |
+        HalPaginationLinks bevat de link properties die nodig zijn om te kunnen navigeren in een collectie:
+
+        - first: uri voor het opvragen van de eerste pagina van deze collectie
+        - previous: uri voor het opvragen van de vorige pagina van deze collectie
+        - next: uri voor het opvragen van de volgende pagina van deze collectie
+      example:
+        self:
+          href: /resourcenaam?page=4
+        first:
+          href: /resourcenaam?page=1
+          title: Eerste pagina
+        previous:
+          href: /resourcenaam?page=3
+          title: Vorige pagina
+        next:
+          href: /resourcenaam?page=5
+          title: Volgende pagina
+      allOf:
+      - $ref: '#/components/schemas/HalCollectionLinks'
+      - type: object
+        properties:
+          first:
+            $ref: '#/components/schemas/HalLink'
+          previous:
+            $ref: '#/components/schemas/HalLink'
+          next:
+            $ref: '#/components/schemas/HalLink'
+    HalCollectionLinks:
+      type: object
+      properties:
+        self:
+          $ref: '#/components/schemas/HalLink'
+      description: |
+        HalCollectionLinks bevat de self link die elke HAL Resource minimaal moet hebben in zijn _links property
+    geometryGeoJSON:
```

Comment:
Hier wordt gebruik gemaakt van de `oneOf` constructie. Dit werkt nog niet bij alle code generatoren. Met NSwag kan ik geen compileerbare code genereren. Met OpenAPITools lukt het wel om compileerbare code te genereren, maar ik weet niet of de code correct gaat deserialiseren. Dat moet ik verder onderzoeken.
@strijm, heb je al code gegenereerd met de yaml bestanden? Ik ben benieuwd of je met jullie java code generator code kan genereren.

Reply:
Los van code generatie, als ik naar deze Geo schema component definities kijk, dan vraag ik mij af of het handig is om deze definities te gebruiken. Ze volgen naar mijn mening niet de OAS specificaties van OAS. In zowel v3.0.3 als in v3.1 wordt aangegeven om de discriminator object te gebruiken bij het gebruik van oneOf

Reply:
Zie: https://spec.openapis.org/oas/v3.1.0#discriminator-object

Reply:
Los van codegeneratie vraag ik mij af of we deze Geo schema definities moeten gebruiken. Naar mijn mening voldoen ze niet aan de OAS 3 specs. Daarin wordt aangegeven om de discriminator object te gebruiken wanneer oneOf wordt gebruikt. Zie: https://spec.openapis.org/oas/v3.1.0#discriminator-object

Reply:
Alhoewel ik het met je eens ben dat het gebruik van de discriminator het gebruik van de oneOf  verduidelijkt is deze volgens de OAS3 specs niet verplicht. Ook bij de 3.1.0 uitleg staat "can be". 

Wel hebben wij vooralsnog in onze Haal Centraal design decisions staan dat we de de oneOf constructie niet toepassen i.v.m. problemen met code-generatie (zie https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/master/docs/design_decisions.md#dd54-gebruik-geen-oneof-of-anyof-constructies-voor-polymorfe-gegevens ).

Reply:
Ik heb nu een Geometrie type gedefinieerd die een cirkel van type Cirkel (allOf van point plus radius) of polygoon kan bevatten. Misschien niet de mooiste oplossing maar wel functioneel en levert geen code generatie problemen op. Evt. kan de cirkel nog gewijzigd worden naar: bevat point en radius.
@JohanBoer @fsamwel @MelvLee is dit werkbaar?

Reply:
Ik heb ook een attribuut locatie van type Point toegevoegd, zodat er nu een keuze is tussen, punt, cirkel of polygoon.